packages/launch/src/launch/index.tsx

```diff
@@ -32,14 +35,24 @@ const FocusedLaunchModal: React.FunctionComponent< Props > = ( {
 	openCheckout,
 	redirectTo,
 	getCurrentLaunchFlowUrl,
+	isInIframe,
+	isLaunchImmediately,
 } ) => {
-	const {
-		isModalDismissible,
-		isModalTitleVisible,
-		shouldDisplaySuccessView,
-	} = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
+	const { isModalDismissible, isModalTitleVisible } = useSelect( ( select ) =>
+		select( LAUNCH_STORE ).getState()
+	);
+	const { launchSite } = useDispatch( SITE_STORE );
+
+	const { closeFocusedLaunch, unsetPlanProductId } = useDispatch( LAUNCH_STORE );
 
-	const { closeFocusedLaunch } = useDispatch( LAUNCH_STORE );
+	React.useEffect( () => {
+		if ( isLaunchImmediately ) {
+			// if there was a plan in cart before redirect to payment processing,
+			// remove it now since we don't need to reload the page when dismissing Success view
+			unsetPlanProductId();
+			launchSite( siteId );
+		}
+	}, [ isLaunchImmediately, unsetPlanProductId, launchSite, siteId ] );
```

Comment:
Context: This one happens when user enters the block editor again after checkout redirection (e.g. wp-admin or paypal).

Reply:
At the moment we're using this only in Calypso (iframed editor) for cases of external payment (eg: Paypal). If we'll be doing something like this also from wp-admin we need to control via some query argument also where the user lands from `/checkout` if they abandon which right now is on `/plans` and should be the editor with launch modal open but not launching immediately.